lib/shared/widgets/custom_navigation_drawer.dart

```diff
@@ -249,16 +289,49 @@ class CustomNavigationDrawer extends StatelessWidget {
         ),
       );
 
-  void _checkToken(
+  Future<void> _checkToken(
     BuildContext context, {
     required String route,
-  }) {
+  }) async {
     final auth = context.read<AuthCubit>();
     final isExpired = auth.state.session.isExpired;
     if (!isExpired) {
       context.goNamed(route);
       return;
     }
+    if (!await LocalAuthInfo.instance.canCheckBiometrics) {
+      // ignore: use_build_context_synchronously
+      if (!context.mounted) {
+        return;
+      }
+      _passwordLogin(context, auth: auth, route: route);
+      return;
+    }
+    try {
+      await LocalAuthInfo.instance.authenticate().then((hasAuthenticated) {
```

Comment:
isn't it a bit confusing to use both async approaches at the same time? I mean both 'await' and 'then'